cumulus_library/base_utils.py

```diff
@@ -1,18 +1,43 @@
 """ Collection of small commonly used utility functions """
 
+import dataclasses
 import datetime
 import json
-import os
 import pathlib
 import shutil
 import zipfile
 from contextlib import contextmanager
 
 from rich import progress
 
+from cumulus_library import databases
 
-def filepath(filename: str) -> str:
-    return os.path.join(os.path.dirname(__file__), filename)
+
+@dataclasses.dataclass
+class StudyConfig:
+    """Class for containing study-centric parameters
+
+    The intent of this class is that if you want something passed through to the
+    prepare_queries section of a study, this is the place it should go. If you're
+    doing something above that level, consider explicit arguments instead. This should
+    be an interface aimed at a study author.
+
+    :param db_backend: a databaseBackend object for a specific target database
+    :keyword db_type: the argument passed in from the CLI indicating the requested DB
+        (this is easier to use in things like jinja templates than db_backend, if they
+        need to run DB technology aware queries)
```

Comment:
i agree, but this ends up being another bit of interface breaking. The only place this is actually used is in the jinja environment template, which means we would need to pass this down through at least some, and potentially all, of those template interface wrappers, which breaks the study API. This would go in the 3.0 changelist.

Reply:
ah, i see what you're saying - i was thinking of where it would actually be used, i.e. in passing to functions in base_templates. I'll make this change.